json/pickler/src/main/scala/sttp/tapir/json/Pickler.scala

```diff
@@ -0,0 +1,359 @@
+        override lazy val writer = summon[Writer[Map[K, V]]]
+        override lazy val reader = summon[Reader[Map[K, V]]]
+      },
+      newSchema
+    )
+
+  inline given picklerForAnyVal[T <: AnyVal]: Pickler[T] = ${ picklerForAnyValImpl[T] }
+
+  private inline def errorForType[T](inline template: String): Unit = ${ errorForTypeImpl[T]('template) }
+
+  private def errorForTypeImpl[T: Type](template: Expr[String])(using Quotes): Expr[Unit] = {
+    import quotes.reflect.*
+    val templateStr = template.valueOrAbort
+    val typeName = TypeRepr.of[T].show
+    report.error(String.format(templateStr, typeName))
+    '{}
+  }
+
+  private def picklerForAnyValImpl[T: Type](using quotes: Quotes): Expr[Pickler[T]] =
+    import quotes.reflect.*
+    val tpe = TypeRepr.of[T]
+
+    val isValueCaseClass =
+      tpe.typeSymbol.isClassDef && tpe.classSymbol.get.flags.is(Flags.Case) && tpe.baseClasses.contains(Symbol.classSymbol("scala.AnyVal"))
+
+    if (!isValueCaseClass) {
+      '{ nonMirrorPickler[T] }
+    } else {
+
+      val field = tpe.typeSymbol.declaredFields.head
+      val fieldTpe = tpe.memberType(field)
+      fieldTpe.asType match
+        case '[f] =>
+          val basePickler = Expr.summon[Pickler[f]].getOrElse {
+            report.errorAndAbort(
+              s"Cannot summon Pickler for value class ${tpe.show}. Missing Pickler[${fieldTpe.show}] in implicit scope."
+            )
+          }
+          '{
+            val newSchema: Schema[T] = ${ basePickler }.schema.as[T]
+            new Pickler[T](
+              new TapirPickle[T] {
+                override lazy val writer = summonInline[Writer[f]].comap[T](
+                  // writing object of type T means writing T.field
+                  ccObj => ${ Select.unique(('ccObj).asTerm, field.name).asExprOf[f] }
+                )
+                // a reader of type f (field) will read it and wrap into value object using the consutructor of T
+                override lazy val reader = summonInline[Reader[f]]
+                  .map[T](fieldObj => ${ Apply(Select.unique(New(Inferred(tpe)), "<init>"), List(('fieldObj).asTerm)).asExprOf[T] })
+              },
+              newSchema
+            )
+          }
+    }
+
+  private inline def fromExistingSchemaAndRw[T](schema: Schema[T])(using ClassTag[T], Configuration, Mirror.Of[T]): Pickler[T] =
+    Pickler(
+      new TapirPickle[T] {
+        val rw: ReadWriter[T] = summonFrom {
+          case foundTapirRW: ReadWriter[T] =>
+            foundTapirRW
+          case foundUpickleDefaultRW: _root_.upickle.default.ReadWriter[T] => // there is BOTH schema and ReadWriter in scope
+            foundUpickleDefaultRW.asInstanceOf[ReadWriter[T]]
+          case _ =>
+            errorForType[T](
+              "Found implicit Schema[%s] but couldn't find a uPickle ReadWriter for this type. Either provide a ReadWriter, or remove the Schema from scope and let Pickler derive its own."
```

Comment:
in the "future tasks" we should also add to include the derivation path up to this point (if possible)